lib/array.js

```diff
@@ -201,6 +204,59 @@ const reduce = (items, fn, done, initial) => {
   fn(previous, current, next, count, items);
 };
 
+const REDUCE_RIGHT_EMPTY_ARR =
+  'Metasync: reduceRight of empty array with no initial value';
+
+// Asynchronous reduceRight
+//   items - <Array>, incoming
+//   fn - <Function>, to be executed for each value in array
+//     previous - <any>, value previously returned in the last iteration
+//     current - <any>, current element being processed in the array
+//     callback - <Function>, callback for returning value
+//         back to reduce function
+//       err - <Error> | <null>
+//       data - <any>, resulting value
+//     counter - <number>, index of the current element
+//         being processed in array
+//     items - <Array>, the array reduce was called upon
+//   done - <Function>, on done, optional
+//     err - <Error> | <null>
+//     result - <Array>
+//   initial - <any>, optional value to be used as first
+//       argument in first iteration
+const reduceRight = (items, fn, done, initial) => {
+  done = done || common.emptyness;
+  const len = items.length;
+  let count = typeof initial === 'undefined' ? len - 2 : len - 1;
```

Comment:
What about array with one element and no initial value? `count` would equal `-1`.

Reply:
What a sharp eye! Many thanks for that :100: 
I will fix it here, fix reduce and add more tests for both functions

Reply:
Done